### PR TITLE
validate_secure_password

### LINKED ISSUE #3: validate_secure_password/2
`Authority.Ecto.Changeset.validate_secure_password/2` would validate that a password meets minimum security requirements. Presumably, these would include length and symbol checks.

```elixir
# Check that the password meets agreed-upon password strength checks
validate_secure_password(changeset, :password)
```

The checks could be configurable, or we could just assume the "best" ones and let you fall back on the built in `Ecto` functions if you don't like it.

Original: infinitered/authority#11

### PR DESCRIPTION
**Pull request by rzane**
Resolves #3.

### CODE REVIEW

File: lib/authority_ecto/changeset.ex [hunk 1]
```diff
@@ -7,8 +7,23 @@ defmodule Authority.Ecto.Changeset do
 
   @type field :: atom
 
+  @password_blacklist File.cwd!()
+                      |> Path.join("password_blacklist.txt")
+                      |> File.read!()
+                      |> String.split("\n")
```
Comment (danielberkompas) [inline lib/authority_ecto/changeset.ex]: This file should probably go in `priv`, and I think you can read it this way:

```elixir
@password_blacklist "priv/password_blacklist.txt"
                    |> File.stream!
                    |> Stream.map(&String.trim/1)
                    |> Enum.into([])
```

It's slightly more efficient.

**Review by danielberkompas — COMMENTED** [review]
Looks good to me so far! I'm reading the NIST guidelines this afternoon; I'll comment I find anything more we should add.

**Comment by danielberkompas** [comment]
This regular expression can be used to exclude repeated characters like "aaaaa": `(.)\1{2}` 

To detect sequences, it looks like we'll need to write code. Elixir binary pattern matching, here we come!

**Comment by rzane** [comment]
Binary matching could definitely work here, but I think this might be simpler:

```elixir
  @consecutive_alphanum "abcdefghijklmnopqrstuvwxyz01234567890"

  defp consecutive?(value, size) do
    blacklist = chunk_letters(@consecutive_alphanum, size)
    value |> chunk_letters(size) |> Enum.any?(&(&1 in blacklist))
  end

  defp chunk_letters(value, size) do
    value |> String.to_charlist |> Enum.chunk_every(size, 1, :discard)
  end
```

**Comment by danielberkompas** [comment]
Yeah, I [wrote a function using pattern matching](https://gist.github.com/danielberkompas/4dc179ae657faebb1be4575d53daf581) but yours is better.

We can use your function to exclude passwords with >= 3 consecutive characters.

```elixir
def validate_consecutive(changeset, field, opts \\ []) do
  value = get_change(changeset, field)
  max = opts[:max] || 3
  msg = opts[:message] || "contains more than %{max} consecutive characters"
  
  if value && consecutive?(value, max) do
    add_error(changeset, field, msg, [validation: :consecutive, max: max])
  else
    changeset
  end 
end
```

File: lib/authority_ecto/password.ex [hunk 2]
```diff
@@ -0,0 +1,34 @@
+defmodule Authority.Ecto.Password do
```
Comment (rzane) [inline lib/authority_ecto/password.ex:1]: perhaps this should be marked `@moduledoc false`?

Comment (danielberkompas) [reply lib/authority_ecto/password.ex:1]: Yes, I agree with that.

**Comment by danielberkompas** [comment]
We add the `@moduledoc false` tag when we do #6.

**Pull request merged**